unity-sdk/AndroidPlugin/OptimoveUnityPlugin/src/main/java/com/optimove/unity/plugin/OptimoveInitProvider.java

```diff
@@ -11,12 +11,19 @@
 
 import com.optimove.android.Optimove;
 import com.optimove.android.OptimoveConfig;
+import com.optimove.android.optimobile.OptimoveInApp;
 
 public class OptimoveInitProvider extends ContentProvider {
     @Override
     public boolean onCreate() {
         Application app = (Application) this.getContext().getApplicationContext();
-        Optimove.initialize(app, new OptimoveConfig.Builder("YOUR OPTIMOVE CREDENTIALS", "YOUR OPTIMOVE MOBILE CREDENTIALS").build());
+        OptimoveConfig.Builder configBuilder = new OptimoveConfig.Builder("YOUR_OPTIMOVE_CREDENTIALS", "YOUR_OPTIMOVE_MOBILE_CREDENTIALS");
+        configBuilder.enableInAppMessaging(OptimoveConfig.InAppConsentStrategy.AUTO_ENROLL);
+        Optimove.initialize(app, configBuilder.build());
+        Optimove.getInstance().pushRegister();
```

Comment:
This init provider has a few differences with the one in cordova
1. we shouldnt call pushRegister, it's called by the consumers of the SDK when appropriate
2. We should override install info
3. We should check inApp strategy and register inApp handler accordingly

Reply:
about the pushRegister - got it, I will change it (I used it for testing)
about override install info, I added the method but I lack knowledge about the correct data we should override with, SDK type, run time version etc, so I'm leaving this now with "garbage" data, about inApp strategy I've added a simple check, which we will expand in conifg Injection for android PR